GVFS/GVFS.Common/Git/GitProcess.cs

```diff
@@ -538,7 +538,9 @@ public Result PackObjects(string filenamePrefix, string gitObjectsDirectory, Act
         /// </summary>
         public Result WriteCommitGraph(string objectDir, List<string> packs)
         {
-            string command = "commit-graph write --stdin-packs --split --size-multiple=4 --object-dir \"" + objectDir + "\"";
+            // Do not expire commit-graph files that have been modified in the last hour.
+            // This will prevent deleting any commit-graph files that are currently in the commit-graph-chain.
+            string command = $"commit-graph write --stdin-packs --split --size-multiple=4 --expire-time={1 * 60 * 60} --object-dir \"{objectDir}\"";

```

Comment:
So if I understand correctly all the current commit-graph-chain files will be "touched" with this `commit-graph write` command? and setting the `--expire-time` will make it not consider those files for delete?  Is there any significance to the 1 hour?  Is that because the files could have been touched but the code to check the expire runs much later and a bunch of time could have passed?

Reply:
No significance to the one hour, except that it is definitely longer than the `git commit-graph write` command. The logic that expires first touches all the files in the chain then the next step is to compare the paths in the `objects/info/commit-graphs` dir to those in the `commit-graph-chain` file. Probably one second would be enough, but let's set the expire time to be relative to how often we expect the maintenance job to run (we prefetch about every 75 minutes, triggering a post-fetch job.)